### ISSUE TITLE
Client-side video editing (MediaBlob)

### ISSUE THREAD

**Issue by ykh015** [w3ctag/design-reviews#514]
Hello TAG! 

  

I'm requesting a TAG review of **Client-side video editing (MediaBlob)**

  
MediaBlob is a high-level API that enables video editing on the browser.  This API solves these common overheads/workarounds that web developers currently use for editing media: 
1. Use external JavaScript/WebAssembly libraries usually in several MBs (affects bandwidth and may not be efficient) 
2. Server roundtrip (affects both time and bandwidth) 

This API is introducing trim, concatenation and split operations on the media files. 

  - Explainer: https://github.com/WICG/video-editing/blob/master/readme.md 

  - GitHub repo (if you prefer feedback filed there): https://github.com/WICG/video-editing/ 

  - Primary contacts: 

      - Yash Khandelwal (@ykh015), Microsoft 

      - Steve Becker (@SteveBeckerMSFT), Microsoft 

  - Organization/project driving the design: Microsoft  

  - External status/issue trackers for this feature (publicly visible, e.g. Chrome Status):  None yet 

  
Further details: 

  - [x] I have reviewed the TAG's [API Design Principles](https://w3ctag.github.io/design-principles/) 

  - The group where the incubation/design work on this is being done (or is intended to be done in the future): WICG 

  - The group where standardization of this work is intended to be done ("unknown" if not known): unknown 

  - Major unresolved issues with or opposition to this design: None yet 

  

We'd prefer the TAG provide feedback as: 

  🐛 open issues in our GitHub repo for **each point of feedback**

**Comment by kenchris** [comment]
What is the relationship to WebCodec? Would it be possible to implement this API client-side on top, for instance as a polyfill/library, or do you foresee performance issues with that approach?

The API does seem quite user friendly, but I wonder if it is flexible enough as some of the use-cases seem to overlap with WebCodec.

Information about the above would be great to add to the explainer

**Comment by kenchris** [comment]
According to my above comment, there seems to be a similar discussion on the blink-dev mailing list:

https://groups.google.com/a/chromium.org/forum/#!searchin/blink-dev/mediablob%7Csort:date/blink-dev/3eac-HVygFY/_BBjNckXBQAJ

**Comment by palemieux** [comment]
As discussed at https://github.com/WICG/video-editing/issues/13 and as currently specified, the API introduces inefficiencies and generational loss in even simple cases. Refactoring the API to be playlist-centric would address these issues and simply future extensions.

**Comment by atanassov** [comment]
One of the strengths with the API is its ergonomics for developers. Further, one of the common scenarios it addresses well can be thought of as: playback > seek > trim > playback. Can you add a more complete example in the explainer that shows how developers will be achieving this? 

@ykh015, there is an active discussion on blink-dev https://github.com/w3ctag/design-reviews/issues/514#issuecomment-641124322, can you summarize what are the next steps for this API?

**Comment by kenchris** [comment]
> Can you add a more complete example in the explainer that shows how developers will be achieving this?

for instance if I am trimming a video stream, I want to be able to seek through the stream (playback) and then when I trim, be able to easily update the UI

**Comment by ykh015** [comment]
After the discussion on the [i2p](https://groups.google.com/a/chromium.org/forum/#!topic/blink-dev/3eac-HVygFY), it was concluded that demuxing and muxing can be handled by JS libraries efficiently and need not be provided by the browser. WebCodecs is already providing  transcoding and when combined together with external JS libraries, media editing scenarios can be addressed.

On providing a more ergonomic and direct API for video editing, we are re-evaluating the current proposed model of using MediaBlobOperation and looking into a playlist type model as being discussed in this [issue](https://github.com/WICG/video-editing/issues/13).

**Comment by palemieux** [comment]
@kenchris What does the _propose closing_ label mean and why was it applied?

**Comment by cynthia** [comment]
> What does the propose closing label mean and why was it applied?

Propose closing generally means "we are done-ish with the review and we'll discuss at our plenary meeting and close if everyone is happy with the outcome of the review". This isn't always the case, but covers about 80% of the times we use that label.

**Comment by kenchris** [comment]
We propose closing as this doesn't seem to be moving forward in its current form and we need to clean our backlog. You can always reopen when you have something new or file another review issue.
